### PR TITLE
FIX, DOC: Bug in plot_evoked_joint API

### LINKED ISSUE #13157: evoked.plot_joint API `exclude` parameter docstring is inconsistent with the code
### Description of the problem

The [API doc](https://mne.tools/1.8/generated/mne.Evoked.html#mne.Evoked.plot_joint) Shows that the default value for parameter `exclude` is `'bads'`, but in the description for that parameter, it says that the default is `None`, when In fact, `None` is not even a valid argument for this parameter:

>     exclude : None | list of str | 'bads'
>        Channels names to exclude from being shown. If ``'bads'``, the
>       bad channels are excluded. Defaults to ``None``.

### Steps to reproduce

```Python
sample_fpath = mne.datasets.sample.data_path()
fpath = sample_fpath / "MEG" / "sample" / "sample_audvis_raw.fif"
raw = mne.io.read_raw(fpath, preload=True).pick("eeg")

epochs = mne.make_fixed_length_epochs(raw, duration=1, preload=True)
evoked = epochs.average()
evoked.plot_joint(exclude=None) # include/exclude must be list, tuple, ndarray, or "bads". You provided type <class 'NoneType'>.
```

### Expected results

Figure shows with no bad channels excluded.

### Actual results

`ValueError: include/exclude must be list, tuple, ndarray, or "bads". You provided type <class 'NoneType'>.`

### Additional information

Should the code be fixed to match the docstring (i.e. accept `None`, and make it the default), or should the mention of `None` simply by removed from the docstring?

### PR DESCRIPTION
**Pull request by scott-huberty**
Fixes #13157 

 FYI It's a little more complicated that I thought...

The _function_ [plot_evoked_joint](https://mne.tools/stable/generated/mne.viz.plot_evoked_joint.html#mne.viz.plot_evoked_joint) does have the default `exclude=None`, **which actually does not work!**  🚨  But the docstring was technically inline with the ~API~ function signature:

https://github.com/mne-tools/mne-python/blob/df4a3f1601418bf1dfd1a3621c34bfa3b1188f8b/mne/viz/evoked.py#L1794-L1857

The _method_ [evoked.plot_joint](https://mne.tools/stable/generated/mne.Evoked.html#mne.Evoked.plot_joint) has a default `exclude="bads"`. The docstring is copied from the _function_ `plot_evoked_joint` to the _method_ `evoked.plot_joint` which is where the docstring-to-code discrepancy arises:

https://github.com/mne-tools/mne-python/blob/8f83332cac8b0cdbc28ca038ef4a66c6f876dc9c/mne/evoked.py#L772-L792

**TL;DR**

We can update the docstring but we have to change

https://github.com/mne-tools/mne-python/blob/df4a3f1601418bf1dfd1a3621c34bfa3b1188f8b/mne/viz/evoked.py#L1800

to `exclude="bads"`

### CODE REVIEW

File: mne/viz/evoked.py [hunk 1]
```diff
@@ -1824,9 +1824,10 @@ def plot_evoked_joint(
         axes are passed make sure to set ``title=None``, otherwise some of your
         axes may be removed during placement of the title axis.
     %(picks_all)s
-    exclude : None | list of str | 'bads'
+    exclude : list of str | 'bads'
         Channels names to exclude from being shown. If ``'bads'``, the
-        bad channels are excluded. Defaults to ``None``.
+        bad channels are excluded. If an empty list, no channels are
```
Comment (scott-huberty) [inline mne/viz/evoked.py:1829]: should we change this to something like `exclude : empty list | list of str | 'bads'`?

File: mne/viz/evoked.py [hunk 2]
```diff
@@ -1824,9 +1824,10 @@ def plot_evoked_joint(
         axes are passed make sure to set ``title=None``, otherwise some of your
         axes may be removed during placement of the title axis.
     %(picks_all)s
-    exclude : None | list of str | 'bads'
+    exclude : list of str | 'bads'
```
Comment (scott-huberty) [inline mne/viz/evoked.py:1827]: should we change this to something like `exclude : empty list | list of str | 'bads'`?

Comment (larsoner) [reply mne/viz/evoked.py:1827]: I don't think so `empty list` is covered by `list of str` well enough I think

**Review by larsoner — COMMENTED** [review]
Should we add a test that `exclude=None` raises an error? Should it?

**Comment by scott-huberty** [comment]
superseded by  #13391
